### PR TITLE
ir_leds: rework timer setup 

### PR DESCRIPTION
**Pull request by jonny-str**
update of MASTER_TIMER triggers CAMERA_TRIGGER_TIMER, LED_850NM_TIMER and LED_940NM_TIMER
Setup of the slave timers:
- fixed prescaler with 1 us per tick
- PWM_mode 2 (0 -> 1 at CC event)
- set  CCR to start delay value
- set ARR to on_time_us + start delay
- single shot mode without retrigger

change ir_camera_timer_settings:
- remove 740nm LEDs
- make naming more clear

delay camera trigger by 70us (only diamond) to account fordelay in constant current driver
Remove 740nm support
Add 850 center, side and 940 single for diamond

Tests: 
- accessibility of diamond and pearl for different IR groups
- if max on-time and duty limits don't trigger the safety circuit

fixes O-2601

### CODE REVIEW

**Comment by jonny-str** [comment]
O-2601

**Comment by moritz-diller** [comment]
Fixes O-2186 too

**Comment by moritz-diller** [comment]
@fouge I tested it on Pearl and Diamond and ready for review now.

File: main_board/src/optics/ir_camera_system/unit_tests/ir_camera_timer_settings/main.c [hunk 1]
```diff
@@ -442,53 +305,50 @@ ZTEST(timer_settings_on_time,
...
-    // same should apply for setting the on-time at 1 fps
-    ret = timer_settings_from_on_time_us(on_time_us, &ts, &ts);
-    zassert_equal(RET_SUCCESS, ret, "");
-    zassert_equal(ts.on_time_in_us, on_time_us, "must be on_time_us");
-    zassert_equal(ts.fps, fps, "must be %u, actual %u", fps, ts.fps);
-    zassert_not_equal(0, ts.psc, "must not be 0, actual %u", ts.psc);
-    zassert_not_equal(0, ts.arr, "must not be 0, actual %u", ts.arr);
-    zassert_equal(1, ts.ccr, "must be 1, actual %u", ts.ccr);
-}
+// ZTEST(timer_settings_on_time,
```
Comment (fouge) [inline main_board/src/optics/ir_camera_system/unit_tests/ir_camera_timer_settings/main.c:316]: should this test be removed?

Comment (jonny-str) [reply main_board/src/optics/ir_camera_system/unit_tests/ir_camera_timer_settings/main.c:316]: yes, it doesnt make sence with the new timer structure, as the ontime timers are independent from the fps timer

Comment (jonny-str) [reply main_board/src/optics/ir_camera_system/unit_tests/ir_camera_timer_settings/main.c:316]: I'll remove it

File: main_board/src/main.c [hunk 2]
```diff
@@ -43,8 +43,6 @@ static K_MUTEX_DEFINE(analog_and_i2c_mutex);
 
 #ifdef CONFIG_ZTEST_NEW_API
 #include <zephyr/ztest.h>
-
-ZTEST_SUITE(hil, NULL, NULL, NULL, NULL, NULL);
```
[the remark this answers is not in the document]
Comment (jonny-str) [reply main_board/src/main.c:47]: what do you sugesst, I wanted to call hil_test_after_t() after each test

File: main_board/src/optics/ir_camera_system/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    |   Timer 3 (940NM Timer)                                                         |
+    |         ┌────────────────────────────────────┐                                  |
+    |    _____|                 Pulse              |______________________________    |
+    |   |<CCR>|                                                                       |
+    |   |<----------------------ARR--------------->|                                  |
+    |                                                                                 |
+    |   Timer 8 / 20 (Camera Trigger Timer)                                           |
+    |             ┌────────────────────────────────┐                                  |
+    |    _________|                 Pulse          |______________________________    |
+    |   |<CCR>|                                                                       |
```
Comment (jonny-str) [inline main_board/src/optics/ir_camera_system/README.md:119]: CCR arrow should be as long as the low pulse, I will change

File: main_board/src/optics/ir_camera_system/ir_camera_timer_settings.h [hunk 4]
```diff
@@ -45,17 +45,31 @@ BUILD_ASSERT(DT_PROP(DT_NODELABEL(rcc), apb1_prescaler) ==
              "If this is not the case, use different macros for APB1 and APB2 "
              "prescaler values.");
 #endif
-#define TIMER_COUNTER_WIDTH_BITS 16
+#define TIMER_COUNTER_WIDTH_BITS    16
+#define IR_CAMERA_SYSTEM_IR_LED_PSC (TIMER_CLOCK_FREQ_MHZ - 1) // 1us per tick
+
+#if defined(CONFIG_BOARD_DIAMOND_MAIN)
+#define IR_LED_TURN_ON_DELAY_US 70
```
Comment (jonny-str) [inline main_board/src/optics/ir_camera_system/ir_camera_timer_settings.h:52]: I measured on the B3 hardware only 36 to 44 us, so I will change this ti 40us

File: main_board/src/optics/ir_camera_system/ir_camera_system_tests.c [hunk 5]
```diff
@@ -322,8 +327,310 @@ send_msg(McuMessage *msg)
...
+        InfraredLEDs_Wavelength_WAVELENGTH_940NM_SINGLE;
+    send_msg(&msg);
+    // should fail with INVALID_PARAM no changes should be observed
+    k_msleep(SEPARATION_TIME_MS);
+    zassert_equal(ir_camera_system_get_enabled_leds(),
+                  InfraredLEDs_Wavelength_WAVELENGTH_940NM);
+#endif
+}
+
+ZTEST(hil, test_invalid_ir_wavelengths_functions)
```
Comment (fouge) [inline main_board/src/optics/ir_camera_system/ir_camera_system_tests.c:535]: I removed this function, duplicate as seen above

File: main_board/src/optics/ir_camera_system/ir_camera_system_tests.c [hunk 6]
```diff
@@ -322,8 +327,310 @@ send_msg(McuMessage *msg)
...
+                          "safety circuit tripped but shouldn't at %d fps with "
+                          "%d us on-time, wavelength %d",
+                          fps,
+                          1e6 / fps * IR_CAMERA_SYSTEM_MAX_IR_LED_DUTY_CYCLE,
+                          wavelength);
+        }
+    }
+}
+
+ZTEST(hil, test_invalid_ir_wavelengths_msgs)
```
Comment (fouge) [inline main_board/src/optics/ir_camera_system/ir_camera_system_tests.c:463]: I removed this function, duplicate

**Review by fouge — APPROVED** [review]

**Pull request merged**